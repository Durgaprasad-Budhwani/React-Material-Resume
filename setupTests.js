/**
 * Created by Durgaprasad Budhwani on 12/14/2017.
 */
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Element.prototype.getBoundingClientRect = function(){
    return {
        y: -100,
        top: -100
    }
};

configure({ adapter: new Adapter() });