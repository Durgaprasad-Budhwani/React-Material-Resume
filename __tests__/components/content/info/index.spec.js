/**
 * Created by durgaprasad on 10/23/17.
 */
/* global expect,it */
import React from 'react';
import InfoContainer, {Info} from '../../../../src/components/content/info';
import renderer from 'react-test-renderer';
import {mount} from 'enzyme'
import toJson from 'enzyme-to-json';
import '../../../../setupTests';

describe('<Info />', () => {
    it('renders <InfoContainer />', () => {
        const tree = renderer.create(
            <InfoContainer/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
    it('renders <Info /> handle openTab event', () => {
        const classes = {
            container: {},
            info: {},
            button: {},
        };
        const wrapper = mount(
            <Info classes={classes}/>
        );
        const instance = wrapper.instance();
        instance._openTab("google");
        expect(toJson(wrapper)).toMatchSnapshot();
    });
})