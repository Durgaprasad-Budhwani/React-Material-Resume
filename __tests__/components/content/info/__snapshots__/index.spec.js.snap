// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Info /> renders <Info /> handle openTab event 1`] = `
<Info
  classes={
    Object {
      "button": Object {},
      "container": Object {},
      "info": Object {},
    }
  }
>
  <div
    className={Object {}}
  >
    <WithStyles(Button)
      className={Object {}}
      color="contrast"
      onClick={[Function]}
      variant="outlined"
    >
      <Button
        className={Object {}}
        classes={
          Object {
            "colorInherit": "MuiButton-colorInherit-25",
            "contained": "MuiButton-contained-15",
            "containedPrimary": "MuiButton-containedPrimary-16",
            "containedSecondary": "MuiButton-containedSecondary-17",
            "disabled": "MuiButton-disabled-24",
            "extendedFab": "MuiButton-extendedFab-22",
            "fab": "MuiButton-fab-21",
            "flat": "MuiButton-flat-9",
            "flatPrimary": "MuiButton-flatPrimary-10",
            "flatSecondary": "MuiButton-flatSecondary-11",
            "focusVisible": "MuiButton-focusVisible-23",
            "fullWidth": "MuiButton-fullWidth-29",
            "label": "MuiButton-label-5",
            "mini": "MuiButton-mini-26",
            "outlined": "MuiButton-outlined-12",
            "outlinedPrimary": "MuiButton-outlinedPrimary-13",
            "outlinedSecondary": "MuiButton-outlinedSecondary-14",
            "raised": "MuiButton-raised-18",
            "raisedPrimary": "MuiButton-raisedPrimary-19",
            "raisedSecondary": "MuiButton-raisedSecondary-20",
            "root": "MuiButton-root-4",
            "sizeLarge": "MuiButton-sizeLarge-28",
            "sizeSmall": "MuiButton-sizeSmall-27",
            "text": "MuiButton-text-6",
            "textPrimary": "MuiButton-textPrimary-7",
            "textSecondary": "MuiButton-textSecondary-8",
          }
        }
        color="contrast"
        component="button"
        disableFocusRipple={false}
        disabled={false}
        fullWidth={false}
        mini={false}
        onClick={[Function]}
        size="medium"
        type="button"
        variant="outlined"
      >
        <WithStyles(ButtonBase)
          className="MuiButton-root-4 MuiButton-outlined-12"
          component="button"
          disabled={false}
          focusRipple={true}
          focusVisibleClassName="MuiButton-focusVisible-23"
          onClick={[Function]}
          type="button"
        >
          <ButtonBase
            centerRipple={false}
            className="MuiButton-root-4 MuiButton-outlined-12"
            classes={
              Object {
                "disabled": "MuiButtonBase-disabled-31",
                "focusVisible": "MuiButtonBase-focusVisible-32",
                "root": "MuiButtonBase-root-30",
              }
            }
            component="button"
            disableRipple={false}
            disableTouchRipple={false}
            disabled={false}
            focusRipple={true}
            focusVisibleClassName="MuiButton-focusVisible-23"
            onClick={[Function]}
            tabIndex="0"
            type="button"
          >
            <button
              className="MuiButtonBase-root-30 MuiButton-root-4 MuiButton-outlined-12"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiButton-label-5"
              >
                Download Resume
              </span>
              <NoSsr
                defer={false}
                fallback={null}
              >
                <WithStyles(TouchRipple)
                  center={false}
                  innerRef={[Function]}
                >
                  <TouchRipple
                    center={false}
                    classes={
                      Object {
                        "child": "MuiTouchRipple-child-37",
                        "childLeaving": "MuiTouchRipple-childLeaving-38",
                        "childPulsate": "MuiTouchRipple-childPulsate-39",
                        "ripple": "MuiTouchRipple-ripple-34",
                        "ripplePulsate": "MuiTouchRipple-ripplePulsate-36",
                        "rippleVisible": "MuiTouchRipple-rippleVisible-35",
                        "root": "MuiTouchRipple-root-33",
                      }
                    }
                  >
                    <TransitionGroup
                      childFactory={[Function]}
                      className="MuiTouchRipple-root-33"
                      component="span"
                      enter={true}
                      exit={true}
                    >
                      <span
                        className="MuiTouchRipple-root-33"
                      />
                    </TransitionGroup>
                  </TouchRipple>
                </WithStyles(TouchRipple)>
              </NoSsr>
            </button>
          </ButtonBase>
        </WithStyles(ButtonBase)>
      </Button>
    </WithStyles(Button)>
    <p
      className={Object {}}
    >
      Hello! I am Durgaprasad Budhwani. Full Stack Developer specializing in web, desktop, mobile design and development, I maintain a healthy balance between functionality and visual impact in all my work. I believe every piece of design can be a work of art. 
    </p>
  </div>
</Info>
`;

exports[`<Info /> renders <InfoContainer /> 1`] = `
<div
  className="Info-container-1"
>
  <button
    className="MuiButtonBase-root-30 MuiButton-root-4 MuiButton-outlined-12 Info-button-3"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-5"
    >
      Download Resume
    </span>
    <span
      className="MuiTouchRipple-root-33"
    />
  </button>
  <p
    className="Info-info-2"
  >
    Hello! I am Durgaprasad Budhwani. Full Stack Developer specializing in web, desktop, mobile design and development, I maintain a healthy balance between functionality and visual impact in all my work. I believe every piece of design can be a work of art. 
  </p>
</div>
`;
